spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/SubmissionControllerHelperService.java

```diff
@@ -238,6 +241,34 @@ public JSONObject downloadSubmissionMetadata(HttpSession session, JSONObject jso
     }
   }
 
+  private static void zipFiles(Set<File> files, File outpath) throws IOException {
+    // Create a buffer for reading the files
+    byte[] buf = new byte[1024];
+
+    ZipOutputStream out = new ZipOutputStream(new FileOutputStream(outpath));
+
+    // Compress the files
+    for (File f : files) {
+      FileInputStream in = new FileInputStream(f);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "FileInputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

Reply:
:+1: